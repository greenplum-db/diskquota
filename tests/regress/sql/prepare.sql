CREATE EXTENSION diskquota;
-- start_ignore
\! gpstop -u
-- end_ignore
SELECT pg_sleep(1);
\! cp data/csmall.txt /tmp/csmall.txt
SELECT pg_sleep(15);

-- prepare a schema that has reached quota limit
CREATE SCHEMA badquota;
DROP ROLE IF EXISTS testbody;
CREATE ROLE testbody;
CREATE TABLE badquota.t1(i INT);
ALTER TABLE badquota.t1 OWNER TO testbody;
INSERT INTO badquota.t1 SELECT generate_series(0, 100000);
SELECT diskquota.init_table_size_table();
SELECT diskquota.set_schema_quota('badquota', '1 MB');
SELECT pg_sleep(10);
SELECT size, segid from diskquota.table_size where tableid in (select oid from pg_class where relname='t1');
-- expect fail
INSERT INTO badquota.t1 SELECT generate_series(0, 10);
