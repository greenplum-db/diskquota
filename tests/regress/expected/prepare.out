CREATE EXTENSION diskquota;
-- start_ignore
\! gpstop -u
20211213:00:02:48:240132 gpstop:laptop:v-[INFO]:-Starting gpstop with args: -u
20211213:00:02:48:240132 gpstop:laptop:v-[INFO]:-Gathering information and validating the environment...
20211213:00:02:48:240132 gpstop:laptop:v-[INFO]:-Obtaining Greenplum Master catalog information
20211213:00:02:48:240132 gpstop:laptop:v-[INFO]:-Obtaining Segment details from master...
20211213:00:02:48:240132 gpstop:laptop:v-[INFO]:-Greenplum Version: 'postgres (Greenplum Database) 6.18.2+dev.173.g55557f44f3 build dev'
20211213:00:02:48:240132 gpstop:laptop:v-[INFO]:-Signalling all postmaster processes to reload
-- end_ignore
SELECT pg_sleep(1);
 pg_sleep 
----------
 
(1 row)

\! cp data/csmall.txt /tmp/csmall.txt
SELECT pg_sleep(15);
 pg_sleep 
----------
 
(1 row)

-- disable hardlimit feature.
SELECT diskquota.disable_hardlimit();
 disable_hardlimit 
-------------------
 
(1 row)

-- prepare a schema that has reached quota limit
CREATE SCHEMA badquota;
DROP ROLE IF EXISTS testbody;
NOTICE:  role "testbody" does not exist, skipping
CREATE ROLE testbody;
NOTICE:  resource queue required -- using default resource queue "pg_default"
CREATE TABLE badquota.t1(i INT);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'i' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
ALTER TABLE badquota.t1 OWNER TO testbody;
INSERT INTO badquota.t1 SELECT generate_series(0, 100000);
SELECT diskquota.init_table_size_table();
 init_table_size_table 
-----------------------
 
(1 row)

SELECT diskquota.set_schema_quota('badquota', '1 MB');
 set_schema_quota 
------------------
 
(1 row)

SELECT pg_sleep(10);
 pg_sleep 
----------
 
(1 row)

SELECT size, segid FROM diskquota.table_size
  WHERE tableid IN (SELECT oid FROM pg_class WHERE relname='t1')
  ORDER BY segid DESC;
  size   | segid 
---------+-------
 1310720 |     2
 1310720 |     1
 1310720 |     0
 3932160 |    -1
(4 rows)

-- expect fail
INSERT INTO badquota.t1 SELECT generate_series(0, 10);
ERROR:  schema's disk space quota exceeded with name:badquota
